lib/streamlit/runtime/secrets.py

```diff
@@ -45,33 +44,47 @@ def _missing_key_error_message(key: str) -> str:
     )
 
 
-class AttrDict(UserDict):  # type: ignore[type-arg]
+class AttrDict(Mapping[str, Any]):
     """
     We use AttrDict to wrap up dictionary values from secrets
     to provide dot access to nested secrets
     """
 
+    def __init__(self, value):
+        self.__dict__.update(value)
```

Comment:
I'm not sure if we should use `__dict__` here. It's probably better to actually create a new variable, e.g. `_dict` or `_data` to manage the dictionary. Using `__dict__` can lead to unexpected issues, since this contains all the attributes of the given object. E.g. if you have a secret called `__len__`.

Reply:
See the implementation of Secrets here:

https://github.com/streamlit/streamlit/blob/f7a8acc257cb248b595b880b2227c6a524955c18/lib/streamlit/runtime/secrets.py#L98

Reply:
Hey, yes!

I just created a separate item `__nested_secrets__` in `__dict__`` 
Because using the same approach with just `self._nested_secrets` will not work since we override `__setattr__` (I did it for custom exception text).

The important note here is that similar to secrets first level lookup

`st.secrets._secrets` or `st.secrets.some_key.__nested_secrets__` will return the dict content instead of the value from `toml file`.

Reply:
Ahh I see, it looks a bit hacky.. but I don;t think there is any problem with doing it this way.